packages/core/src/core.ts

```diff
@@ -1159,7 +1167,11 @@ export class McpServer {
    * duplicates are silently skipped.
    * @internal
    */
-  private mountChild(prefix: string, suffix: string, child: McpServer): void {
+  private mountChild(
+    prefix: string,
+    suffix: string,
+    child: McpServer<TConfig>,
```

Comment:
thinking about the implication for child/grouped servers. all their Config types have to be compatible?